db/tikv/coprocessor.go

```diff
@@ -74,17 +73,70 @@ func (db *coprocessor) Read(ctx context.Context, table string, key string, field
 	if err != nil {
 		return nil, err
 	}
-	//return db.table.DecodeValue(result.GetData(), fields)
+	//return db.table.DecolsdeValue(result.GetData(), fields)
 	return make(map[string][]byte, len(fields)), nil
 }
 
 func (db *coprocessor) Scan(ctx context.Context, table string, startKey string, count int, fields []string) ([]map[string][]byte, error) {
-
-	return nil, errors.New("unsupported type of scan in coprocessor")
+	// construct the req
+	req := kv.Request{}
+	req.Concurrency = 1
+	req.Tp = kv.ReqTypeDAG
+	req.KeyRanges = []kv.KeyRange{db.table.GetScanRange(startKey)}
+	dag := db.table.BuildDAGTableScanWithLimitReq(fields, count)
+	req.StartTs = dag.StartTs
+	data, err := dag.Marshal()
+	if err != nil {
+		return nil, err
+	}
+	req.Data = data
+	// send req
+	client := db.db.GetClient()
+	res := client.Send(ctx, &req, nil)
+	defer res.Close()
+	_, err = res.Next(ctx)
```

Comment:
Next only returns one item?

Reply:
Next returns marshaled [SelectResponse](https://github.com/pingcap/tipb/blob/master/proto/select.proto#L28:9) which contains some rows.
I have added the loop for calling the next until the nil is returned.